### PR TITLE
SC-22333 getting profile usertype and location details from new fields

### PR DESCRIPTION
**Pull request by Hari-stackroute**
In Self-declared report locations and usertype and use to get from old fields usertype, usersubtype and locationids.
Now new fields are added in user tables profilelocation and profileusertype, so those details should be read from this new fields.

### Type of change

Please choose appropriate options.

- [x] New feature (non-breaking change which adds functionality)

### How Has This Been Tested?

Please describe the tests that you ran to verify your changes in the below checkboxes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Existing test cases fits the changes

### CODE REVIEW

**Review by reshmi-nair — APPROVED** [review]

File: data-products/src/main/scala/org/sunbird/analytics/job/report/StateAdminReportJob.scala [hunk 1]
```diff
@@ -114,6 +118,37 @@ object StateAdminReportJob extends optional.Application with IJob with StateAdmi
...
+                val profileUserType = JSONUtils.deserialize[Map[String, String]](f.getString(0))
+                val usertypeList = profileUserType.filter(f => f._1.equalsIgnoreCase("type")).map(f => f._2)
+                usertype = if (!usertypeList.isEmpty) usertypeList.head else ""
+                val usersubtypeList = profileUserType.filter(f => f._1.equalsIgnoreCase("subType")).map(f => f._2)
+                usersubtype = if (!usersubtypeList.isEmpty) usersubtypeList.head else ""
+            }
+            var locations = new ListBuffer[String]()
+            if (null != f.getString(1) && f.getString(1).nonEmpty) {
+                val loc = JSONUtils.deserialize[List[Map[String, String]]](f.getString(1))
+                val stateIdList = loc.filter(f => f.getOrElse("type", "").equalsIgnoreCase("state")).map(f => f.getOrElse("id", ""))
```
Comment (SanthoshVasabhaktula) [inline data-products/src/main/scala/org/sunbird/analytics/job/report/StateAdminReportJob.scala:136]: Use UDF functions on dataframe

**Review by SanthoshVasabhaktula — CHANGES_REQUESTED** [review]

File: data-products/src/main/scala/org/sunbird/analytics/job/report/StateAdminReportJob.scala (hunk 1 above)
Comment (Hari-stackroute) [reply data-products/src/main/scala/org/sunbird/analytics/job/report/StateAdminReportJob.scala]: @SanthoshVasabhaktula : done the changes please check.

File: data-products/src/main/scala/org/sunbird/analytics/job/report/StateAdminReportJob.scala [hunk 2]
```diff
@@ -83,37 +86,51 @@ object StateAdminReportJob extends optional.Application with IJob with StateAdmi
...
     
+    def decryptPhoneEmailInDF(userDF: DataFrame, email: String, phone: String)(implicit sparkSession: SparkSession, fc: FrameworkContext) : DataFrame = {
+        val profileEmailMap = userDF.rdd.map(r => (r.getAs[String]("userid"), r.getAs[String](email))).collectAsMap()
+        val profilePhoneMap = userDF.rdd.map(r => (r.getAs[String]("userid"), r.getAs[String](phone))).collectAsMap()
+        val decryptedUserProfileDF = decryptDF(profileEmailMap, profilePhoneMap);
+        decryptedUserProfileDF
+    }
+    
+    def appendUserProfileTypeWithLocation(userDf: DataFrame) : DataFrame = {
+        userDf.show(10,false)
```
Comment (SanthoshVasabhaktula) [inline data-products/src/main/scala/org/sunbird/analytics/job/report/StateAdminReportJob.scala:126]: remove .show()

File: data-products/src/main/scala/org/sunbird/analytics/job/report/StateAdminReportJob.scala [hunk 3]
```diff
@@ -83,37 +86,51 @@ object StateAdminReportJob extends optional.Application with IJob with StateAdmi
...
+        val userProfileDf = userDf.withColumn("locationids", locationIdList(col("profilelocation"))).
+            withColumn("usertype", addUserType(col("profileusertype"))).
+            withColumn("usersubtype", addUserSubType(col("profileusertype")))
+        userProfileDf.show(10,false)
```
Comment (SanthoshVasabhaktula) [inline data-products/src/main/scala/org/sunbird/analytics/job/report/StateAdminReportJob.scala:130]: remove .show()

**Review by SanthoshVasabhaktula — CHANGES_REQUESTED** [review]

File: data-products/src/main/scala/org/sunbird/analytics/job/report/StateAdminReportJob.scala (hunk 2 above)
Comment (Hari-stackroute) [reply data-products/src/main/scala/org/sunbird/analytics/job/report/StateAdminReportJob.scala]: @SanthoshVasabhaktula done

File: data-products/src/main/scala/org/sunbird/analytics/job/report/StateAdminReportJob.scala (hunk 3 above)
Comment (Hari-stackroute) [reply data-products/src/main/scala/org/sunbird/analytics/job/report/StateAdminReportJob.scala]: @SanthoshVasabhaktula done

**Review by SanthoshVasabhaktula — APPROVED** [review]

**Pull request merged**